communication/http/set-http-header.yml

```diff
@@ -4,11 +4,13 @@ rule:
     namespace: communication/http
     authors:
       - michael.hunhoff@mandiant.com
+      - anushka.virgaonkar@mandiant.com
     scope: function
     mbc:
       - Communication::HTTP Communication::Set Header [C0002.013]
     examples:
       - 6A352C3E55E8AE5ED39DC1BE7FB964B1:0x1000E230
   features:
-    - and:
+    - or:
       - api: winhttp.WinHttpAddRequestHeaders
+      - property: System.Net.WebClient::Headers
```

Comment:
can we distinguish get vs. set of a property? if not, should we limit this to one rule?

Reply:
Per discussion from https://github.com/mandiant/capa/issues/939 we've opted not to differentiate get v. set although I believe we would have all of the information to do so. Curious to @anushkavirgaonkar 's thoughts on the our ability to differentiate between get v. set using .NET metadata.

The alternative to differentiating get v. set is to label all property rules as "access" which I think is limiting and forces use to create more rules than ideally necessary. I prefer adding the ability to differentiate get v. set.

Reply:
The `System.Net.WebClient.Headers` property is being used for both getting and setting a collection of headers. I understand that it would mean that both the rules `set http header` and `read http header` would be matched when the property is referenced in the sample. Since we are not distinguishing between get vs set, should I keep the rules as they are (and we'd have to be okay with having a false positive) or should I remove the rules for maintaining accuracy?

Reply:
Mike raises a good point. @anushkavirgaonkar do you think it's worth adding the differentiation? Would you be able to get this feature in?
From what we learned with other features I think it makes sense to add it from the start, it's more powerful, and we can extract it "easily". We'd need to agree on a good syntax though.

Reply:
Sure! Differentiating between read and write sounds like a good idea. I've added commits to https://github.com/mandiant/capa/pull/1122 for emitting the features and have updated the rules here as well. I've followed the option 2 for the rule syntax from here: https://github.com/mandiant/capa/issues/939#issuecomment-1210406078

Reply:
Thanks, this is great!